### PR TITLE
Prepare run-tests.sh for podman

### PR DESCRIPTION
**Pull request by tyll**
The first commit allows to run multiple containers at once without error messages regarding the docker networks. The other commits also prepare run-tests.sh to be used with podman.

---
This change is [Reviewable](https://reviewable.io/reviews/nmstate/nmstate/577)

### CODE REVIEW

File: automation/run-tests.sh [hunk 1]
```diff
@@ -281,34 +276,34 @@ if [[ "$CI" == "true" ]];then
...
 
-docker_exec '(source /etc/os-release; echo $PRETTY_NAME); rpm -q NetworkManager'
+container_exec '(source /etc/os-release; echo $PRETTY_NAME); rpm -q NetworkManager'
 
 dump_network_info
 
 pyclean
 if [[ "$CI" != "true" ]];then
-    docker_exec "cp -rf $CONTAINER_WORKSPACE /tmp/"
+    container_exec "cp -rf $CONTAINER_WORKSPACE /tmp/"
```
Comment (cathay4t) [inline automation/run-tests.sh]: In podman, the `/tmp` is mounted as `no_exec`. Please use another folder.

File: automation/run-tests.sh [hunk 2]
```diff
@@ -18,27 +18,29 @@ TEST_TYPE_INTEG="integ"
 FEDORA_IMAGE_DEV="nmstate/fedora-nmstate-dev"
 CENTOS_IMAGE_DEV="nmstate/centos8-nmstate-dev"
 
+: ${container_cmd:=docker}
```
Comment (cathay4t) [inline automation/run-tests.sh]: Please use capital for environment variable

Comment (tyll) [reply automation/run-tests.sh]: will change this

File: automation/run-tests.sh (hunk 1 above)
Comment (tyll) [reply automation/run-tests.sh]: Changed it to use `/root/nmstate-workspace`.

**Comment by tyll** [comment]
In pratice, it does not yet work completely with podman because the `preserve_old_config` fixture at https://github.com/nmstate/nmstate/blob/master/tests/integration/conftest.py#L59 does not work with the `tap0` interface that is used in the podman container.

**Comment by tyll** [comment]
@cathay4t I addressed your concerns, can you please re-check?

File: automation/run-tests.sh [hunk 3]
```diff
@@ -21,40 +18,38 @@ TEST_TYPE_INTEG="integ"
...
 function add_extra_networks {
-    docker network create $NET0 || true
-    docker network create $NET1 || true
-    docker network connect $NET0 $CONTAINER_ID
-    docker network connect $NET1 $CONTAINER_ID
-    docker_exec '
+    container_exec '
+      ip link add eth1 type veth peer eth1peer && \
+      ip link add eth2 type veth peer eth2peer && \
       ip addr flush eth1 && \
```
Comment (EdDev) [inline automation/run-tests.sh]: These are no longer needed now.

File: automation/run-tests.sh [hunk 4]
```diff
@@ -21,40 +18,38 @@ TEST_TYPE_INTEG="integ"
...
 }
 
 function add_extra_networks {
-    docker network create $NET0 || true
-    docker network create $NET1 || true
-    docker network connect $NET0 $CONTAINER_ID
-    docker network connect $NET1 $CONTAINER_ID
-    docker_exec '
+    container_exec '
+      ip link add eth1 type veth peer eth1peer && \
```
Comment (EdDev) [inline automation/run-tests.sh]: Don’t you need to set ‘link up’ on them? At least for the peer ones which we do not suppose to manage using nm.

File: automation/run-tests.sh (hunk 3 above)
Comment (tyll) [reply automation/run-tests.sh]: removed them

File: automation/run-tests.sh (hunk 4 above)
Comment (cathay4t) [reply automation/run-tests.sh]: Yeah. `ip link up` is required.

**Comment by cathay4t** [comment]
I was hoping(PR #441):
 * veth creation could be done by pytest fixture.
 * `run-tests.sh` kicks of pytest/tox commands with preferred arguments.
 * `run-tests-container.sh` setup containers and invoke `run-tests.sh`.

But basically, I am OK with current approach, just a suggestion.

File: automation/run-tests.sh (hunk 4 above)
Comment (tyll) [reply automation/run-tests.sh]: Changed it for the peer. It is passing with NM master in the CI, not sure why it fails for 1.20.

**Comment by tyll** [comment]
> I was hoping(PR #441):
> 
> * veth creation could be done by pytest fixture.

This might also be an option. We can still build upon this PR to do this eventually. It would be nice if pytest would still allow to run it against externally created interfaces, for example when testing it in VMs against virtual interfaces.

> * `run-tests.sh` kicks of pytest/tox commands with preferred arguments.

The preferred arguments should probably be moved from run-tests.sh to tox, so that tox always have the correct arguments. Not sure if there is something special for the CI that needs to be different, though.

> * `run-tests-container.sh` setup containers and invoke `run-tests.sh`.

This would only be needed if special arguments besides the ones from tox are needed AFAICS. However, it would be nice to split the script to do the image setup in an Travis install phase to make the log display more useful (making it show the test output only in the normal case). It can also still be done.

**Comment by tyll** [comment]
@cathay4t can you maybe revisit this? The docker error 129 with cp is driving me nuts and with #588 it should be possible to run podman locally.

**Comment by tyll** [comment]
@cathay4t travis passed on all platforms, can you please approve this PR?

**Pull request merged**